### PR TITLE
Fresh 539

### PR DESCRIPTION
**Pull request by metas-rc**

### CODE REVIEW

File: de.metas.acct.base/src/main/java/de/metas/acct/api/IDocumentBL.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+	/**
+	 * Retrieve all the documents that are marked as posted but do not actually have fact accounts
+	 * Exclude the documents with no fact accounts that were not supposed to be posted (always 0 in posting)
+	 * 
+	 * @param ctx
+	 * @param startTime
+	 * @return
+	 */
+	List<Object> retrievePostedWithoutFactActt(Properties ctx, Timestamp startTime);
```
Comment (metas-ts) [inline de.metas.acct.base/src/main/java/de/metas/acct/api/IDocumentBL.java]: why not return DocAction? (maybe i'll see further down)

File: de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * #L%
+ */
+
+/**
+ * The documents (created one day before) that were marked as posted but have no fact accounts will be reposted by this process
+ * 
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+public class Documents_FactAcct_Creation_For_Posted extends SvrProcess
```
Comment (metas-ts) [inline de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java]: we need an option to look back further that one day. Please add an optional date parameter for that.
You should be able to use the `de.metas.process.Param` annotation

File: de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+		final IPostingService postingService = Services.get(IPostingService.class);
+
+		for (final Object document : documentsPostedNoFacts)
+		{
+			final ITableRecordReference tableRecordRef = TableRecordReference.of(document);
+
+			// Note: Do not change this message!
+			// The view de_metas_acct.Reposted_Documents is based on it.
+			loggable.addLog("Document Reposted: AD_Table_ID = {}, Record_ID = {}.", tableRecordRef.getAD_Table_ID(), tableRecordRef.getRecord_ID());
```
Comment (metas-ts) [inline de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java]: Please also output the documents table name and documentn number, if it is a docaction.
Also check out my remarks at the view's DDL

File: de.metas.acct.base/src/main/sql/postgresql/ddl/views/RV_Fact_Acct_Reposted_Documents.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+pinst.AD_Client_ID,
+pinst.AD_Org_ID,
+pinst.Created,
+pinst.CreatedBy,
+pinst.Updated,
+pinst.UpdatedBy,
+pinst.IsActive,
+
+pinst.AD_PInstance_ID,
+to_number (regexp_matches(p_msg, '\d+')::text, '99999999999999999') AS AD_Table_ID ,
```
Comment (metas-ts) [inline de.metas.acct.base/src/main/sql/postgresql/ddl/views/RV_Fact_Acct_Reposted_Documents.sql]: After having checked https://www.postgresql.org/docs/9.1/static/functions-matching.html, i propose using substring() instead. e.g.
```
SELECT 
	substring('Document Reposted: AD_Table_ID=123, Record_ID = 456.' FROM 'AD_Table_ID[ =]*([0-9]*)')::numeric AS AS_AD_Table_ID,
	substring('Document Reposted: AD_Table_ID=123, Record_ID = 456.' FROM 'Record_ID[ =]*([0-9]*)')::numeric AS Record_ID
```

File: de.metas.acct.base/src/main/java/de/metas/acct/api/IDocumentBL.java (hunk 1 above)
Comment (metas-rc) [inline de.metas.acct.base/src/main/java/de/metas/acct/api/IDocumentBL.java]: Done

File: de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java (hunk 3 above)
Comment (metas-rc) [inline de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java]: Done

File: de.metas.acct.base/src/main/sql/postgresql/ddl/views/RV_Fact_Acct_Reposted_Documents.sql (hunk 4 above)
Comment (metas-rc) [inline de.metas.acct.base/src/main/sql/postgresql/ddl/views/RV_Fact_Acct_Reposted_Documents.sql]: Done.
Also added the where clause 
_AND substring(p_msg FROM 'AD_Table_ID[ =]*([0-9]*)')!='' AND substring(p_msg FROM 'Record_ID[ =]*([0-9]*)')!='' to the view

File: de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java (hunk 2 above)
Comment (metas-rc) [inline de.metas.acct.base/src/main/java/de/metas/acct/process/Documents_FactAcct_Creation_For_Posted.java]: Done

**Pull request merged**